oat-sa/extension-tao-community
# Feature/ADF-1418/Migrate sass

### Related to: [ADF-1417](https://oat-sa.atlassian.net/browse/ADF-1417) and [ADF-1418](https://oat-sa.atlassian.net/browse/ADF-1418)

Requires: 

### Summary

Update the stylesheets using proper SASS functions.

BREAKING CHANGE:
- The package now requires Node.js 14.17+
- Replace Node-Sass with Dart-Sass

### Details

To properly support Node 18, the Node-SASS needs to be replaced by Dart-SASS.
This lib is the official version of SASS, and it has breaking changes for the set of supported features (see [SASS: breaking changes](https://sass-lang.com/documentation/breaking-changes/extend-compound)).
Among others:
- compound selectors are no longer supported
- division is not supported outside of `calc`
- color computation must pass through the color functions see https://github.com/sass/sass/issues/2543

### How to test
- install the extension
- check the style on TAO's backoffice

[ADF-1417]: https://oat-sa.atlassian.net/browse/ADF-1417?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[ADF-1418]: https://oat-sa.atlassian.net/browse/ADF-1418?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
